src/components/ui/Box.tsx

```diff
@@ -51,6 +52,14 @@ export interface BoxProps {
    * Otherwise leave this unset so the shifting is done if the label is outside the viewport.
    */
   parentDivRef?: React.RefObject<HTMLElement>;
+  /**
+   * progress bar for the box, replaces the bottom left and bottom right
+   */
+  progress?: {
+    label?: string;
+    percentage: number;
+    type: ProgressType;
+  };
```

Comment:
This component may be getting a bit overloaded. There's possibly a better way to do this, or just display the progress in general.